presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1729,7 +1729,10 @@ private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope so
                         if (starPrefix.isPresent()) {
                             throw new SemanticException(MISSING_TABLE, item, "Table '%s' not found", starPrefix.get());
                         }
-                        throw new SemanticException(WILDCARD_WITHOUT_FROM, item, "SELECT * not allowed in queries without FROM clause");
+                        if (!node.getFrom().isPresent()) {
+                            throw new SemanticException(WILDCARD_WITHOUT_FROM, item, "SELECT * not allowed in queries without FROM clause");
+                        }
+                        throw new SemanticException(COLUMN_NAME_NOT_SPECIFIED, item, "SELECT * not allowed from relation that has no columns");
```

Comment:
That is technically not truth (columns might just be unsupported). It is probably hard to detect it here, but we can make the message more meaningful, e.g:
`SELECT * not allowed from relation that has no columns (are all columns of unsupported type?)`

Reply:
Well, I can change the message. However, the concept of supported/unsupported columns appears to be connector-specific, so I didn't want to mention this in presto main. Also note that `show columns from ...` simply skips unsupported columns (always did), so we're kind of consequently lying that a column does not exist when it is not supported.

Reply:
I agree with keeping the original message since the unsupported part is connector specific.